.github/workflows/add_member.yml

```diff
@@ -28,35 +28,41 @@ jobs:
 
       - name: Checkout code
         uses: actions/checkout@v5
+        with:
+          persist-credentials: false
 
       - name: Get username to add
         id: get_username
         run:
-          python -c "print('USERNAME='+'${{ github.event.issue.title }}'.split(' - ')[1].strip().lstrip('@'))" >> $GITHUB_ENV
+          python -c "print('USERNAME='+'${GITHUB_EVENT_ISSUE_TITLE}'.split(' - ')[1].strip().lstrip('@'))" >> $GITHUB_ENV
+        env:
+          GITHUB_EVENT_ISSUE_TITLE: ${{ github.event.issue.title }}
 
       - name: Validate add user request
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          ISSUE_USER: ${{ github.event.issue.user.login }}
+          ISSUE_USER: ${{ env.ISSUE_USER }}
+          USERNAME: ${{ env.USERNAME }}
+          ISSUE_NUMBER: ${{ env.ISSUE_NUMBER }}
```

Comment:
Something seems off here. Either we don't need to make these re-assignments, or we're no longer capturing the proper data.

Reply:
I'm not sure I understand. What zizmor was pointing out in was that the environment variables could be exposed to injection via shell expansion. By refactoring this was that risk is mitigated. I'm seeing all of the variables being used below ... feels like I'm missing something now 😄

Reply:
This seems correct to me. By makings these reassignments and using them like this, it stops injection attacks.

It does look a bit weird and unnecessary on first glance, but it is absolutely on purpose 😄

Reply:
@Stormheg can you link me to something that explains that? I don't understand why we're putting something into the environment when it's already in the environment.

Reply:
@tim-schilling I think [this part of the zizmor docs](https://docs.zizmor.sh/audits/#template-injection) might help to answer your question